lib/python/picongpu/pypicongpu/output/openpmd.py

```diff
@@ -0,0 +1,138 @@
+# This file is part of PIConGPU.
+# Copyright 2021-2025 PIConGPU contributors
+# Authors: Masoud Afshari
+# License: GPLv3+
+
+from .timestepspec import TimeStepSpec
+from .plugin import Plugin
+from .. import util
+from .openpmd_source import (
+    ChargeDensity,
+    BoundElectronDensity,
+    Counter,
+    Density,
+    Energy,
+    EnergyDensity,
+    EnergyDensityCutoff,
+    LarmorPower,
+    MacroCounter,
+    MidCurrentDensityComponent,
+    Momentum,
+    MomentumDensity,
+    WeightedVelocity,
+)
+import typeguard
+import typing
+
+
+@typeguard.typechecked
+class Source:
+    sources = util.build_typesafe_property(
+        typing.List[
+            typing.Union[
+                str,
+                ChargeDensity,
+                BoundElectronDensity,
+                Counter,
+                Density,
+                Energy,
+                EnergyDensity,
+                EnergyDensityCutoff,
+                LarmorPower,
+                MacroCounter,
+                MidCurrentDensityComponent,
+                Momentum,
+                MomentumDensity,
+                WeightedVelocity,
+            ]
+        ]
+    )
+
+    def __init__(
+        self,
+        sources: typing.List[
+            typing.Union[
+                str,
+                ChargeDensity,
+                BoundElectronDensity,
+                Counter,
+                Density,
+                Energy,
+                EnergyDensity,
+                EnergyDensityCutoff,
+                LarmorPower,
+                MacroCounter,
+                MidCurrentDensityComponent,
+                Momentum,
+                MomentumDensity,
+                WeightedVelocity,
+            ]
```

Comment:
I am not a fan of the many repeats of the all the different source,please switch to using interfaces and testing with isinstance for the interface